frontend/src/components/loginOutButton/index.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useAuth0 } from '../../react-auth0-spa';
+
+// Button to log in user via Auth0
+// The button is disabled when Auth0 authentication is loading
+
+const LoginOutButton = () => {
+  const {
+    loading,
+    isAuthenticated,
+    loginWithRedirect,
+    logout,
+    user,
+  } = useAuth0();
+
+  return (
+    <div className="login-out-button">
+      {!isAuthenticated ? (
+        <button
+          type="button"
+          disabled={loading}
+          onClick={() => loginWithRedirect({})}>
+          Log in
+        </button>
+      ) : (
+        <button
+          type="button"
+          disabled={loading}
+          onClick={() => logout({ federated: true })}>
+          Log out {!loading && user.name}
+        </button>
```

Comment:
There is a little code duplication here. It might be neater to just render 1 button, with the `isAuthenticated ?` check either happening in the onClick method and the button text, or extracted into it's own methods for each. Not sure which you'd prefer or which you think would look better

Reply:
I've had a look and changed this.